dpnp/backend/kernels/dpnp_krnl_logic.cpp

```diff
@@ -71,6 +71,52 @@ void dpnp_all_c(const void* array1_in, void* result1, const size_t size)
     event.wait();
 }
 
+template <typename _DataType1, typename _DataType2, typename _ResultType>
+class dpnp_allclose_c_kernel;
+
+template <typename _DataType1, typename _DataType2, typename _ResultType>
+void dpnp_allclose_c(const void* array1_in, const void* array2_in, void* result1, const size_t size, double rtol_val, double atol_val)
+{
+    if (!array1_in || !result1)
+    {
+        return;
+    }
+
+    const _DataType1* array1 = reinterpret_cast<const _DataType1*>(array1_in);
+    const _DataType2* array2 = reinterpret_cast<const _DataType2*>(array2_in);
+    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
+
+    result[0] = true;
+
+    if (!size)
+    {
+        return;
+    }
+
+    cl::sycl::event event;
+
+    cl::sycl::range<1> gws(size);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0];
+
+        if (std::abs(array1[i] - array2[i]) > (atol_val + rtol_val * std::abs(array2[i]))){
```

Comment:
Just for thought.
`std::abs` works only on host. That means you do memcpy every time between host and GPU dev when you call std::abs. This can have a significant impact on performance.
We have `cl::sycl::fabs` for genfloats, and you can implement your own abs kernel for gen integers.

Reply:
Per my understanding, kernel function is compiled and executed on device as one whole. So it is impossible that part of it is executed on device and another part is executed on host. It means that compiler was able to compile `std::abs` in kernel function.
On another hand should be a reason why `cl::sycl::fabs` exists, possibly its implementation is optimized for kernels. It is a good point to compare performance of implementations of kernel with `std::abs` and with `cl::sycl::fabs`.
Maybe we should create a separate issue/task about it.